libshpool/src/daemon/server.rs

```diff
@@ -659,7 +666,8 @@ impl Server {
             // to avoid breakage and vars the user has asked us to inject.
             .env_clear();
 
-        let term = self.inject_env(&mut cmd, &user_info, header).context("setting up shell env")?;
+        let term = shell_env.iter().filter(|(k, _)| k == "TERM").map(|(_, v)| v).next();
+        cmd.envs(shell_env.to_vec());
```

Comment:
(no action needed, unless this is not correct) It seems like you're creating the env, then injecting it into the shell anyway. I like this separation, nice job

Reply:
yeah